### PR TITLE
Update Secret Volume to support `items`.

### PR DESCRIPTION
**Pull request by sl1pm4t**
This works the same as `config_map.items`.

I also fixed `validateAttributeValueDoesNotContain()` validator. It’s
logic was the reverse of intention.

### CODE REVIEW

**Comment by sl1pm4t** [comment]
```
/usr/local/go/bin/go test -v -timeout 180s -tags  -run ^TestAccKubernetesPod_basic|TestAccKubernetesPod_importBasic|TestAccKubernetesPod_with_pod_security_context|TestAccKubernetesPod_with_container_liveness_probe_using_exec|TestAccKubernetesPod_with_container_liveness_probe_using_http_get|TestAccKubernetesPod_with_container_liveness_probe_using_tcp|TestAccKubernetesPod_with_container_lifecycle|TestAccKubernetesPod_with_container_security_context|TestAccKubernetesPod_with_volume_mount|TestAccKubernetesPod_with_cfg_map_volume_mount|TestAccKubernetesPod_with_resource_requirements|TestAccKubernetesPod_with_empty_dir_volume|TestAccKubernetesPod_with_secret_vol_items$

=== RUN   TestAccKubernetesPod_basic
--- PASS: TestAccKubernetesPod_basic (18.57s)
=== RUN   TestAccKubernetesPod_importBasic
--- PASS: TestAccKubernetesPod_importBasic (4.51s)
=== RUN   TestAccKubernetesPod_with_pod_security_context
--- PASS: TestAccKubernetesPod_with_pod_security_context (5.75s)
=== RUN   TestAccKubernetesPod_with_container_liveness_probe_using_exec
--- PASS: TestAccKubernetesPod_with_container_liveness_probe_using_exec (43.30s)
=== RUN   TestAccKubernetesPod_with_container_liveness_probe_using_http_get
--- PASS: TestAccKubernetesPod_with_container_liveness_probe_using_http_get (20.74s)
=== RUN   TestAccKubernetesPod_with_container_liveness_probe_using_tcp
--- PASS: TestAccKubernetesPod_with_container_liveness_probe_using_tcp (20.43s)
=== RUN   TestAccKubernetesPod_with_container_lifecycle
--- PASS: TestAccKubernetesPod_with_container_lifecycle (6.03s)
=== RUN   TestAccKubernetesPod_with_container_security_context
--- PASS: TestAccKubernetesPod_with_container_security_context (7.93s)
=== RUN   TestAccKubernetesPod_with_volume_mount
--- PASS: TestAccKubernetesPod_with_volume_mount (8.46s)
=== RUN   TestAccKubernetesPod_with_cfg_map_volume_mount
--- PASS: TestAccKubernetesPod_with_cfg_map_volume_mount (7.68s)
=== RUN   TestAccKubernetesPod_with_resource_requirements
--- PASS: TestAccKubernetesPod_with_resource_requirements (5.53s)
=== RUN   TestAccKubernetesPod_with_empty_dir_volume
--- PASS: TestAccKubernetesPod_with_empty_dir_volume (8.14s)
=== RUN   TestAccKubernetesPod_with_secret_vol_items
--- PASS: TestAccKubernetesPod_with_secret_vol_items (8.47s)
PASS
ok  	github.com/terraform-providers/terraform-provider-kubernetes/kubernetes	165.598s
Success: Tests passed.

```

File: kubernetes/schema_pod_spec.go [hunk 1]
```diff
@@ -355,6 +355,42 @@ func volumeSchema() *schema.Resource {
 		MaxItems:    1,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
+				"default_mode": {
+					Type:         schema.TypeInt,
+					Description:  "Optional: mode bits to use on created files by default. Must be a value between 0 and 0777. Defaults to 0644. Directories within the path are not affected by this setting. This might be in conflict with other options that affect the file mode, like fsGroup, and the result can be other mode bits set.",
+					Optional:     true,
+					ValidateFunc: validateModeBits,
```
Comment (radeksimko) [inline kubernetes/schema_pod_spec.go]: Does it make sense to add `Default: 0644` here? Either for clarity for the user when they see the diff and/or for consistency, otherwise the default value (if undefined) here will be `0` which I'm not sure is the same in this context.

File: kubernetes/structures_pod.go [hunk 2]
```diff
@@ -279,9 +279,24 @@ func flattenEmptyDirVolumeSource(in *v1.EmptyDirVolumeSource) []interface{} {
...
+	}
 	if in.SecretName != "" {
 		att["secret_name"] = in.SecretName
 	}
+	if len(in.Items) > 0 {
+		items := make([]interface{}, len(in.Items))
+		for i, v := range in.Items {
+			m := map[string]interface{}{}
+			m["key"] = v.Key
+			m["mode"] = int(*v.Mode)
```
Comment (radeksimko) [inline kubernetes/structures_pod.go]: Should we add a nil check here in case `v.Mode` is `nil` to prevent crash?

File: kubernetes/structures_pod.go [hunk 3]
```diff
@@ -279,9 +279,24 @@ func flattenEmptyDirVolumeSource(in *v1.EmptyDirVolumeSource) []interface{} {
...
+			m["path"] = v.Path
+			items[i] = m
+		}
+		att["items"] = items
+	}
+	att["optional"] = *in.Optional
```
Comment (radeksimko) [inline kubernetes/structures_pod.go]: We can't assume this codepath will only ever process resources created by Terraform. Keep in mind that it's also used for imports - so even if we always pass `Optional` to the API when creating the resource, it doesn't mean it will always come back.

TL;DR we should add a nil check, IMO.

File: kubernetes/resource_kubernetes_pod_test.go [hunk 4]
```diff
@@ -716,6 +742,52 @@ resource "kubernetes_pod" "test" {
...
+      }
+    }
+    volume {
+      name = "db"
+      secret {
+				secret_name = "${kubernetes_secret.test.metadata.0.name}"
+				items {
+					key = "one"
+					path = "path/to/one"
+				}
```
Comment (radeksimko) [inline kubernetes/resource_kubernetes_pod_test.go]: Nitpick: do you mind fixing the indentation here?

File: kubernetes/schema_pod_spec.go [hunk 5]
```diff
@@ -355,6 +355,42 @@ func volumeSchema() *schema.Resource {
...
+				},
+				"items": {
+					Type:        schema.TypeList,
+					Description: `If unspecified, each key-value pair in the Data field of the referenced Secret will be projected into the volume as a file whose name is the key and content is the value. If specified, the listed keys will be projected into the specified paths, and unlisted keys will not be present. If a key is specified which is not present in the Secret, the volume setup will error. Paths must be relative and may not contain the '..' path or start with '..'.`,
```
Comment (radeksimko) [inline kubernetes/schema_pod_spec.go]: Nitpick: Do you mind keeping one-line strings enclosed in double quotes? 🙂

**Comment by radeksimko** [comment]
Thank you for making those changes quickly. This LGTM now. 👍
